peerprep-fe/src/app/(main)/components/filter/TopicsPopover.tsx

```diff
@@ -0,0 +1,103 @@
+'use client';
+import { useState, useEffect } from 'react';
+import { axiosQuestionClient } from '@/network/axiosClient';
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from '@/components/ui/popover';
+import { Button } from '@/components/ui/button';
+import { Check, ChevronsUpDown } from 'lucide-react';
+import { cn } from '@/lib/utils';
+import { Input } from '@/components/ui/input';
+import { ScrollArea } from '@/components/ui/scroll-area';
+
+interface TopicsPopoverProps {
+  selectedTopics: string[];
+  onChange: (value: string[]) => void;
+}
+
+export function TopicsPopover({
+  selectedTopics,
+  onChange,
+}: TopicsPopoverProps) {
+  const [open, setOpen] = useState(false);
+  const [topics, setTopics] = useState<string[]>([]);
+  const [searchTerm, setSearchTerm] = useState('');
+
+  useEffect(() => {
+    const fetchTopics = async () => {
+      try {
+        const response = await axiosQuestionClient.get('/questions/tags');
```

Comment:
This doesnt seem to work yet, i think there isn't an API endpoint for /questions/tags